proto/marginal_state.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+import "mixings.proto";
+
+import "ls_state.proto";
+
+package bayesmix;
+
+
+message MarginalState {
```

Comment:
Why is it called marginal state? I thought it was supposed to only be used for marginal algorithms, but then I saw you used it directly in the base Algorithm class

Reply:
This is a HACK right now, but just to get us started. 
ConditionalState will require saving additional informations, depending on which kind of 'conditional' algorithm we want to implement. In the nearby future we should think of dealing with different States (e.g. by having a template Collector class?) without copy-pasting code too much!